src/DynamicData/Cache/Internal/TransformAsync.cs

```diff
@@ -3,71 +3,60 @@
 // See the LICENSE file in the project root for full license information.
 
 using System.Reactive.Linq;
-
+using System.Reactive.Threading.Tasks;
 using DynamicData.Kernel;
 
 namespace DynamicData.Cache.Internal;
 
-internal sealed class TransformAsync<TDestination, TSource, TKey>(IObservable<IChangeSet<TSource, TKey>> source, Func<TSource, Optional<TSource>, TKey, Task<TDestination>> transformFactory, Action<Error<TSource, TKey>>? exceptionCallback, IObservable<Func<TSource, TKey, bool>>? forceTransform = null)
+internal class TransformAsync<TDestination, TSource, TKey>(
+    IObservable<IChangeSet<TSource, TKey>> source,
+    Func<TSource, Optional<TSource>, TKey, Task<TDestination>> transformFactory,
+    Action<Error<TSource, TKey>>? exceptionCallback,
+    IObservable<Func<TSource, TKey, bool>>? forceTransform = null,
+    int maximumConcurrency = 4,
```

Comment:
The introduction of this parameter is the main performance improvement here, yeah?

I would have the default be `null`, for a couple of reasons:

1. You're really just passing this parameter directly to `.Merge()` and `.Merge()` doesn't enforce a default, you either supply a limit, or there isn't one.
2. 4 is the reasonable default that .NET usually uses for CPU parallelism, but this isn't about limiting CPU parallelism. The practical scenario that this supports is about I/O concurrency, E.G. database queries.

Otherwise, this functionally looks good to me.

Reply:
Yeah, I had no idea there was an overload of Merge that took a max concurrency parameter.  But having looked it up, I agree it should be null, and if it is, then invoke the overload that doesn't use one.

Reply:
I only found out about the overload to limit concurrency recently myself. Rx is the gift that keeps giving.

You both make fair points, and the same also occurred to me. So it will be null.

What do you two think about point 2? Complex overloads are a cognitive pain which optional params tried to resolve. That's why I M thinking about moving towards options objects for primitive optional values

Reply:
Options objects wouldn't be TOO crazy, but I'm not sure where the line is for how many parameters it takes to merit a full object. Is there any wisdom we can glean from Microsoft's usages of that pattern? If we did do this, we would definitely want to implement Options objects as `readonly record struct` I think.

In my personal opinion, I don't consider optional parameters as being cognitive pain at *all*, because when I invoke almost any method where the parameter purpose isn't self-evident at a glance, I do...

```cs
var result = SomeMethod(
    parameter1: value1,
    parameter2: value2,
    parameter3: value3);
```

This plays seamlessly with optional parameters, cause all I have to do is omit the ones I don't want.

This is also pretty much functionally and semantically identical to using an options struct.

```cs
var result = SomeMethod(new()
{
    Option1 = value1,
    Option2 = value2,
    Option3 = value3
});

Both of these offer essentially the exact same cost for the method call, as all the parameters/options (used or not) are allocated on the stack. The difference really is just syntax, and how much code we have to write in the library.

Reply:
`record struct` is fine, but not all versions of C# support it.  Either way, make sure to pass it using `in` so that it isn't copied on the stack.

An options object is not a bad idea because it enables changes to be made without breaking things (adding new options, changing defaults) but we should also have overloads that allow a few commonly used parameters to be set without having to create the object.

Reply:
> it enables changes to be made without breaking things

Oooh, an excellent point.